src/check.rs

```diff
@@ -65,13 +88,59 @@ fn check_file_url(url: &Url, _ctx: &CheckContext) -> Result<(), CheckError> {
 
     if path.is_file() || path.join("index.html").is_file() {
         debug!("Linked file at path {} does exist.", path.display());
-        Ok(())
+        match url.fragment() {
+            Some(fragment) => check_file_fragment(&path, fragment),
+            None => Ok(()),
+        }
     } else {
         debug!("Linked file at path {} does not exist!", path.display());
         Err(CheckError::File(path))
     }
 }
 
+fn check_file_fragment(path: &Path, fragment: &str) -> Result<(), CheckError> {
+    debug!("Checking fragment {} of file {}.", fragment, path.display());
+    let (_, targets) = parse_html_file(path, path);
+
+    if targets.contains(fragment) {
+        Ok(())
+    } else {
+        // Rust documentation uses `#n-m` fragments and JavaScript to highlight
+        // a range of lines in HTML of source code, an element with `id`
+        // attribute of (literal) "#n-m" will not exist, but elements with
+        // `id`s n through m should, this parses the ranged n-m anchor and
+        // checks if elements with `id`s n through m do exist
+        match RUST_LINE_HIGLIGHT_RX.captures(fragment) {
```

Comment:
So, this is the reason I haven't implemented this feature yet - I'd prefer to do this by parsing the HTML fragment, instead of using regex, but the html5ever API is a little painful to use and will be going away soon (#79). I think this should wait for #86 and then go through lol_html.

Reply:
This bit is for fragments that are in form `#n-m` only, for example this one: https://docs.rs/html5ever/0.9.0/src/html5ever/.cargo/registry/src/github.com-1ecc6299db9ec823/html5ever-0.9.0/src/rcdom.rs.html#155-164, an element with id of `#155-164` doesn't exist in that HTML file. What is done here is that on that page a bit of JavaScript kicks in and highlights lines 155 through 164 as elements with ids 155..164 do exist. It's a curiosity of Rustdoc that I needed to implement, otherwise check would fail for integration tests (IIRC `non_existent_http_link` was failing).